Content.Server/Tabletop/TabletopSystem.cs

```diff
@@ -0,0 +1,193 @@
+            RaiseNetworkEvent(new TabletopPlayEvent(table.Uid, camera.Uid, "Chess", (274 + 64, 274)), playerSession.ConnectedClient);
+        }
+
+        /// <summary>
+        /// Create a session associated to this entity UID, if it does not already exist, and return it.
+        /// </summary>
+        /// <param name="uid">The entity UID to ensure a session for.</param>
+        /// <returns>The created/stored tabletop game session.</returns>
+        private TabletopSession EnsureSession(EntityUid uid)
+        {
+            // We already have a session, return it
+            // TODO: if tables are connected, treat them as a single entity
+            if (_gameSessions.ContainsKey(uid))
+            {
+                return _gameSessions[uid];
+            }
+
+            // Session does not exist for this entity yet, create a map and create a session
+            var mapId = _mapManager.CreateMap();
+
+            // Tabletop maps do not need lighting, turn it off
+            var mapComponent = _mapManager.GetMapEntity(mapId).GetComponent<IMapComponent>();
+            mapComponent.LightingEnabled = false;
+            mapComponent.Dirty();
+
+            _gameSessions.Add(uid, new TabletopSession(mapId));
+            var session = _gameSessions[uid];
+
+            // Since this is the first time opening this session, set up the game
+            // TODO: don't assume we're playing chess
+            SetupChessBoard(session.MapId);
+
+            return session;
+        }
+
+        #region Event handlers
+
+        // Move an entity which is dragged by the user, but check if they are allowed to do so and to these coordinates
+        private void OnTabletopMove(TabletopMoveEvent msg, EntitySessionEventArgs args)
+        {
+            if (args.SenderSession as IPlayerSession is not { AttachedEntity: { } playerEntity } playerSession) return;
+
+            // Check if player is actually playing at this table
+            if (!_gameSessions[msg.TableUid].IsPlaying(playerSession)) return;
+
+            // Return if can not see table or stunned/no hands
+            if (!EntityManager.TryGetEntity(msg.TableUid, out var table)) return;
+            if (!CanSeeTable(playerEntity, table) || StunnedOrNoHands(playerEntity)) return;
+
+            // Check if moved entity exists and has tabletop draggable component
+            if (!EntityManager.TryGetEntity(msg.MovedEntityUid, out var movedEntity)) return;
+            if (!movedEntity.HasComponent<TabletopDraggableComponent>()) return;
+
+            // TODO: some permission system, disallow movement if you're not permitted to move the item
+
+            // Move the entity and dirty it (we use the map ID from the entity so noone can try to be funny and move the item to another map)
+            var entityCoordinates = new EntityCoordinates(_mapManager.GetMapEntityId(movedEntity.Transform.MapID), msg.Coordinates.Position);
```

Comment:
Use ComponentManager over EntityManager (so ComponentManager.HasComponent<TabletopDraggableComponent> and ComponentManager.GetComponent<TransformComponent>(uid, out var transform)